### PR TITLE
Code Review Changes

### PR DESCRIPTION
**Pull request by vumd-UC**
Code Review Changes for Sprint 2
Comments for changes available in file level comments

### CODE REVIEW

File: app/src/main/java/edu/uc/brown5a7/BottomsUp/searchAdaptor/drinkAdapter.kt [hunk 1]
```diff
@@ -18,7 +18,7 @@ class drinkAdapter(val context: Context, val drinks: ArrayList<Drink>) : Recycle
 
     inner class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
 
-        var currentdrink: Drink? = null;
+        var currentdrink: Drink? = null
```
Comment (vumd-UC) [inline app/src/main/java/edu/uc/brown5a7/BottomsUp/searchAdaptor/drinkAdapter.kt:21]: Redundant semi-colon in Kotlin. Java practices die hard sometimes.

File: app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkAdapterIntegrationTest.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    private fun thenDrinkCountMatchesItemCount(itemCount : Int)
+    {
+        var result = false
+        if (drinkCount == itemCount)
+        {
+            result = true
+        }
+        assertTrue(result)
+    }
+}
```
Comment (vumd-UC) [inline app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkAdapterIntegrationTest.kt:62]: Sample Integration Test for the drinkAdapter. Given the importance of the drinkAdapter, would not be the worse idea to have an integration test to make sure it works. Conversely, the importance of the drinkAdapter class makes it hard for me to confidently make any changes that would make this integration test work. For that reason, I have committed the file as reference, but have opted to use the @Ignore tag to not mess with CircleCI.

File: app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDTOMockkUnitTest.kt [hunk 3]
```diff
@@ -28,34 +27,21 @@ class DrinkDTOMockkUnitTest {
...
-        confirmVerified(drinkService)
-    }
-
-    private fun givenAFeedOfMockkedDrinkDataAreAvailable() {
+    @Before
+    fun setup()
+    {
         mvm = MainViewModel()
         createMockkData()
     }
```
Comment (vumd-UC) [inline app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDTOMockkUnitTest.kt:35]: This change was in the Sprint1 code review, but didn't quite get carried over. I see it reverted to the original implementation in commit 29a12af1f4c0f219d80b2afc1d718cc7188ea4aa, but I'm not 100% sure if that is intentional. For that reason, I have re-added back in the original changes from the Sprint 1 Code Review.

File: app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDTOMockkUnitTest.kt [hunk 4]
```diff
@@ -28,34 +27,21 @@ class DrinkDTOMockkUnitTest {
...
+        val allDrinks = ArrayList<Drink>()
 
-        //create mockk drink
-        var coke = Drink("26", "soda", "coca-cola", "coke", "Really good on a warm day!", "4")
+        val coke = Drink("26", "soda", "coca-cola", "coke", "Really good on a warm day!", "4")
         allDrinks.add(coke)
-        var MariageFreres = Drink("61", "Tea", "Marco Polo Blue", "Mariage Freres",  "My Favorite French Tea", "5")
-        allDrinks.add(MariageFreres)
+        val mariageFreres = Drink("61", "Tea", "Marco Polo Blue", "Mariage Freres",  "My Favorite French Tea", "5")
+        allDrinks.add(mariageFreres)
```
Comment (vumd-UC) [inline app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDTOMockkUnitTest.kt:44]: As we never changes these values in the test, changed from the var type to the val type.

File: app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDTOMockkUnitTest.kt [hunk 5]
```diff
@@ -64,15 +50,42 @@ class DrinkDTOMockkUnitTest {
...
+        thenResultContainsMariageFreres()
+        thenVerifiedMariageFreresFunctionsEnvoked()
+    }
+
+    @Test
+    fun searchForGarbage_ReturnsNothing(){
+        whenISearchForGarbage()
+        thenIGetZeroResults()
+    }
+
```
Comment (vumd-UC) [inline app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDTOMockkUnitTest.kt:72]: Might be more of a personal preference then a "Best Practice", but organizing all of the tests, givens, whens, and thens makes for a more organized integration test.

File: app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDTOMockkUnitTest.kt [hunk 6]
```diff
@@ -64,15 +50,42 @@ class DrinkDTOMockkUnitTest {
...
+    private fun whenISearchForGarbage() {
+        mvm.fetchDrinks("asdflkj")
+    }
+
     private fun thenResultContainsCoke() {
         var cokeFound = false
 
         mvm.drinks.observeForever {
-            //observe the drink data
```
Comment (vumd-UC) [inline app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDTOMockkUnitTest.kt:75]: Redundant comment in code. Definitely in the camp for "Clean Code" where functions should clearly be named what they do. In this case, observeForever is pretty self evident on what it does.

File: app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDTOMockkUnitTest.kt [hunk 7]
```diff
@@ -98,35 +109,19 @@ class DrinkDTOMockkUnitTest {
         confirmVerified(drinkService)
     }
 
-    private fun whenSearchForMariageFreres() {
-        mvm.fetchDrinks("Mariage Freres")
-    }
-
     private fun thenResultContainsMariageFreres() {
-        var MariageFreresFound = false
+        var mariageFreresFound = false
```
Comment (vumd-UC) [inline app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDTOMockkUnitTest.kt:113]: Changing variable to stay consistent with camel case.

File: app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDTOMockkUnitTest.kt [hunk 8]
```diff
@@ -98,35 +109,19 @@ class DrinkDTOMockkUnitTest {
...
 
         mvm.drinks.observeForever {
-            //observe the drink data
             assertNotNull(it)
             assertTrue(it.size > 0)
             it.forEach {
                 if (it.name.contains("Mariage Freres") && it.brand.contains("Marco Polo Blue") && it.category == "Tea" && it.drinkId == "61" && it.comment.contains("My Favorite French Tea") && it.rating == "5") {
-                    MariageFreresFound = true
+                    mariageFreresFound = true
```
Comment (vumd-UC) [inline app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDTOMockkUnitTest.kt:120]: Changing variable name to stay consistent with camel case.

File: app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDataIntegrationTest.kt [hunk 9]
```diff
@@ -5,10 +5,15 @@ import edu.uc.brown5a7.BottomsUp.service.DrinkService
...
 import org.junit.Assert
+import org.junit.Before
 import org.junit.Rule
 import org.junit.Test
 import org.junit.rules.TestRule
 
+/*
+ * Integration test for Drinkservice.kt. Tests fetching of drinks based on search terms and search categories.
+ */
+
```
Comment (vumd-UC) [inline app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDataIntegrationTest.kt:16]: JavaDoc/KDoc added to test.

**Review by vumd-UC — COMMENTED** [review]
Additional, non file comments:
I would definitely take a look at [this commit](https://github.com/robertdhart99/BottomsUp/commit/29a12af1f4c0f219d80b2afc1d718cc7188ea4aa), as it seems filled with a lot of files that are both unnecessary and computer-specific. Definitely look into using a .gitignore to avoid putting in files that shouldn't necessarily be on the master branch.

File: app/src/main/java/edu/uc/brown5a7/BottomsUp/searchAdaptor/drinkAdapter.kt (hunk 1 above)
Comment (discospiff) [reply app/src/main/java/edu/uc/brown5a7/BottomsUp/searchAdaptor/drinkAdapter.kt:21]: Yes I agree.  :)  When I go back to Java, I forget to put the semicolon in as well.

File: app/src/main/java/edu/uc/brown5a7/BottomsUp/searchAdaptor/drinkAdapter.kt [hunk 10]
```diff
@@ -87,5 +87,5 @@ class drinkAdapter(val context: Context, val drinks: ArrayList<Drink>) : Recycle
         this.drinks = drinks
         drinkListFull = ArrayList(drinks)
     }
-    //
```
Comment (discospiff) [inline app/src/main/java/edu/uc/brown5a7/BottomsUp/searchAdaptor/drinkAdapter.kt:90]: This wasn't doing anything.  Best to remove.

File: app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkAdapterIntegrationTest.kt (hunk 2 above)
Comment (discospiff) [reply app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkAdapterIntegrationTest.kt:62]: Huh... really nice contribution.   Increasing test coverage decreases technical debt.

File: app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDTOMockkUnitTest.kt (hunk 4 above)
Comment (discospiff) [reply app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDTOMockkUnitTest.kt:44]: Yes... this aligns well with Kotlin Coding Conventions.  https://kotlinlang.org/docs/reference/coding-conventions.htm

File: app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDTOMockkUnitTest.kt (hunk 5 above)
Comment (discospiff) [reply app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDTOMockkUnitTest.kt:72]: I agree... readable this way.

File: app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDTOMockkUnitTest.kt (hunk 6 above)
Comment (discospiff) [reply app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDTOMockkUnitTest.kt:75]: Yes... one can read this from the code itself.

File: app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDTOMockkUnitTest.kt (hunk 7 above)
Comment (discospiff) [reply app/src/test/java/edu/uc/brown5a7/BottomsUp/DrinkDTOMockkUnitTest.kt:113]: Yes... this is consistent with convention.

**Comment by discospiff** [comment]
Many really nice contributions here.  This review goes a long way in reducing technical debt.  I recommend merging these changes.

**Pull request merged**